rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -0,0 +1,124 @@
+from rotkehlchen.types import ChecksumEthAddress, EthereumTransaction
+
+
+class ConvexDecoder(DecoderInterface):
+    @staticmethod
+    def _decode_cvxcrv_rewards(
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                if tx_log.topics[0] == CVXCRV_REWARD_PAID_TOPIC or tx_log.topics[0] == CVX_LOCKER_REWARD_PAID_TOPIC:  # noqa: E501
+                    event.event_subtype = HistoryEventSubType.REWARD
+                elif tx_log.topics[0] == CVXCRV_WITHDRAWAL_TOPIC:
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                event.counterparty = CPT_CONVEX
+            elif event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                event.event_type = HistoryEventType.DEPOSIT
+                event.counterparty = CPT_CONVEX
+
+        return None, None
+
+    @staticmethod
+    def _decode_booster(
+            tx_log: EthereumTxReceiptLog,  # pylint: disable=unused-argument
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                if event.counterparty == ZERO_ADDRESS:
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                else:
+                    event.event_type = HistoryEventType.DEPOSIT
+                event.counterparty = CPT_CONVEX
+            elif event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.counterparty = CPT_CONVEX
+        return None, None
+
+    @staticmethod
+    def _decode_cvx(
+            tx_log: EthereumTxReceiptLog,
+            decoded_events: List[HistoryBaseEntry],
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE:  # noqa: E501
```

Comment:
When iterating over the events you should be checking other conditions like the asset matching the one you expect in the event and that the address of the user is the one matching the event location

https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/ethereum/modules/oneinch/v1/decoder.py#L63

This is to prevent errors in bundled transactions

Reply:
Done